core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/internal/validation/ValidationRuleSchedulingRule.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2011-2018 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.core.internal.validation;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import org.eclipse.core.runtime.jobs.IJobManager;
+import org.eclipse.core.runtime.jobs.ISchedulingRule;
+import org.eclipse.core.runtime.jobs.Job;
+
+import com.b2international.snowowl.core.validation.rule.ValidationRule.CheckType;
+
+/**
+ * @since 6.6
+ */
+public class ValidationRuleSchedulingRule implements ISchedulingRule {
+	
+	public static final int MAXIMUM_AMOUNT_OF_CONCURRENT_NORMAL_JOBS = 3;
+	
+	private final CheckType checkType;
+	private final int maxConcurrentJobs;
+	private final String uniqueRuleId;
```

Comment:
What is `uniqueRuleId`? This should be either just `ruleId` or just `id`. IDs are meant to be **unique**.